clar2wasm/src/linker.rs

```diff
@@ -2979,200 +2980,250 @@ fn link_map_delete_fn(linker: &mut Linker<ClarityWasmContext>) -> Result<(), Err
+fn check_height_valid(
+    caller: &mut Caller<'_, ClarityWasmContext>,
+    memory: &Memory,
+    height_lo: i64,
+    height_hi: i64,
+    return_offset: i32,
+) -> Result<u32, Error> {
+    let height = (height_lo as u128) | ((height_hi as u128) << 64);
+
+    let height_value = match u32::try_from(height) {
+        Ok(result) => result,
+        _ => {
+            // Write a 0 to the return buffer for `none`
+            write_to_wasm(
+                caller,
+                *memory,
+                &TypeSignature::BoolType,
+                return_offset,
+                return_offset + get_type_size(&TypeSignature::BoolType),
+                &Value::Bool(false),
+                true,
+            )?;
+            return Err(Error::Wasm(WasmError::ValueTypeMismatch));
+        }
+    };
+
+    let current_block_height = caller
+        .data_mut()
+        .global_context
+        .database
+        .get_current_block_height();
+    if height_value >= current_block_height {
+        // Write a 0 to the return buffer for `none`
+        write_to_wasm(
+            caller,
+            *memory,
+            &TypeSignature::BoolType,
+            return_offset,
+            return_offset + get_type_size(&TypeSignature::BoolType),
+            &Value::Bool(false),
+            true,
+        )?;
+    }
+    Ok(height_value)
+}
+
+/// Link host interface function, `get_block_info_time`, into the Wasm module.
+/// This function is called for the `get-block-info? time` expression.
+fn link_get_block_info_time_property_fn(
+    linker: &mut Linker<ClarityWasmContext>,
+) -> Result<(), Error> {
     linker
         .func_wrap(
             "clarity",
-            "get_block_info",
+            "get_block_info_time_property",
             |mut caller: Caller<'_, ClarityWasmContext>,
-             name_offset: i32,
-             name_length: i32,
              height_lo: i64,
              height_hi: i64,
              return_offset: i32,
              _return_length: i32| {
-                // runtime_cost(ClarityCostFunction::BlockInfo, env, 0)?;
+                let memory = caller
+                    .get_export("memory")
+                    .and_then(|export| export.into_memory())
+                    .ok_or(Error::Wasm(WasmError::MemoryNotFound))?;
 
-                // Get the memory from the caller
+                let height_value =
+                    check_height_valid(&mut caller, &memory, height_lo, height_hi, return_offset)?;
+
+                let block_time = caller
+                    .data_mut()
+                    .global_context
+                    .database
+                    .get_block_time(height_value)?;
+                write_to_wasm(
+                    &mut caller,
+                    memory,
+                    &TypeSignature::UIntType,
+                    return_offset,
+                    return_offset + get_type_size(&TypeSignature::UIntType),
+                    &Value::UInt(block_time as u128),
```

Comment:
Don't you have to wrap the result in an Optional?

Reply:
Yes, it seems that I missed that. 
Thank you!